ibridges/__main__.py

```diff
@@ -0,0 +1,288 @@
+
+
+def main() -> None:
+    """CLI pointing to different entrypoints."""
+    # show help by default, else consume first argument
+    subcommand = "--help" if len(sys.argv) < 2 else sys.argv.pop(1)
+
+    if subcommand in ["-h", "--help"]:
+        print(MAIN_HELP_MESSAGE)
+    elif subcommand in ["-v", "--version"]:
+        print(f"Metasyn CLI version {version('metasyn')}")
+
+    # find the subcommand in this module and run it!
+    elif subcommand == "download":
+        ibridges_download()
+    elif subcommand == "upload":
+        ibridges_upload()
+    elif subcommand == "sync":
+        ibridges_sync()
+    elif subcommand == "init":
+        ibridges_init()
+    elif subcommand == "ls":
+        ibridges_ls()
+    else:
+        print(f"Invalid subcommand ({subcommand}). For help see ibridges --help")
+        sys.exit(1)
+
+def _set_ienv_path(ienv_path: Union[None, str, Path]):
+    try:
+        with open(IBRIDGES_CONFIG_FP, "r", encoding="utf-8") as handle:
+            ibridges_conf = json.load(handle)
+    except FileNotFoundError:
+        if ienv_path is None:
+            return None
+        ibridges_conf = {}
+        IBRIDGES_CONFIG_FP.parent.mkdir(exist_ok=True)
+
+    if ienv_path is not None:
+        ibridges_conf["cli_last_env"] = str(Path(ienv_path).absolute())
+    else:
+        ibridges_conf["cli_last_env"] = None
+
+    with open(IBRIDGES_CONFIG_FP, "w", encoding="utf-8") as handle:
+        json.dump(ibridges_conf, handle)
+    return ibridges_conf["cli_last_env"]
+
+
+def _get_ienv_path() -> Union[None, str]:
+    try:
+        with open(IBRIDGES_CONFIG_FP, "r", encoding="utf-8") as handle:
+            ibridges_conf = json.load(handle)
+            return ibridges_conf.get("cli_last_env")
+    except FileNotFoundError:
+        return None
+
+
+def ibridges_init():
+    """Create a cached password for future use."""
+    parser = argparse.ArgumentParser(
+        prog="ibridges init",
+        description="Cache your iRODS password to be used later."
+    )
+    parser.add_argument(
+        "irods_env_path",
+        help="The path to your iRODS environment JSON file.",
+        type=Path,
+        default=None,
+        nargs="?",
+    )
+    args, _ = parser.parse_known_args()
+    ienv_path = _set_ienv_path(args.irods_env_path)
+    print(ienv_path, args.irods_env_path)
+    with interactive_auth(irods_env_path=ienv_path) as session:
+        assert isinstance(session, Session)
+    print("ibridges init was succesful.")
+
+
+def _list_coll(session: Session, remote_path: IrodsPath):
+    if remote_path.collection_exists():
+        print(str(remote_path)+':')
+        coll = get_collection(session, remote_path)
+        print('\n'.join(['  '+sub.path for sub in coll.data_objects]))
+        print('\n'.join(['  C- '+sub.path for sub in coll.subcollections]))
+    else:
+        raise ValueError("Irods path '{remote_path}}' is not a collection.")
```

Comment:
```suggestion
        raise ValueError(f"Irods path {remote_path} is not a collection.")
```

Reply:
I put the quotes there for readability, do you have a strong preference to remove them?

Reply:
Oh that is fine. But we need the 'f' since now the error is not printed with the correct information.

Reply:
Sorry, missed that part, fixed it now.